M90 changes

TJatPBnJ opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
The m90 is now 14tc instead of 18. Its grenades can be bought from the uplink in boxes for 6tc, instead of individual grenades for 2tc each. You can also fire the m90's grenade launcher with right click. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
The m90 isn't used very often, and the grenade launcher on it is currently hard to use, so having right click fire it makes it less difficult and more intuitive to use. For 18tc, it's overshadowed by the l6, which is why I lowered the cost of the gun and it's grenades.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
tweak: The M90 now uses right click to fire it's grenade launcher.
balance: 40mm grenades are now bought in ammo boxes for 6tc rather than individually for 2tc
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TJatPBnJ committed:
no hexane

bye

TJatPBnJ committed:
the m90 is now better

The M90 costs 14 tc instead of 18, the grenade can be fired using right click, grenades are bought from the uplink in boxes instead of individually, the grenade box now holds 8 grenades instead of 4, and the grenade box now costs 6 tc.

TJatPBnJ committed:
Revert "the m90 is now better"

This reverts commit 568ed359e4e1a104d210959b947c2398bddefdfd.

TJatPBnJ committed:
Revert "no hexane"

This reverts commit fa90cbda9f0905823b3f78f2db3b0070e116e225.

TJatPBnJ committed:
the m90 is now better

The m90 is now 14 tc instead of 18, it's grenades can be bought from the uplink in boxes for 6tc, you can fire the grenades with right click, and the grenade ammo box now holds 8 grenades instead of 4.

TJatPBnJ commented:
any feedback or suggestions would be appreciated

Pugie commented:
God bless

necromanceranne commented:
Right click grenade launchers, absolute genius.

Fhaxaris commented:
![image](https://user-images.githubusercontent.com/13771064/106999858-8dda4700-674c-11eb-9ab5-8b105a23a97f.png)
it doesn't look like they have an icon

TJatPBnJ commented:
hmm i was hoping that didnt do anything
ill just lower it back to 4, 8 does seem a bit much after a bit of thinking

TJatPBnJ committed:
uninvisible

Son-of-Space commented:
A weapon never seen will now be used. Good job.

Jared-Fogle commented on code/modules/projectiles/guns/ballistic/automatic.dm:
```diff
@@ -165,7 +165,8 @@
 /obj/item/gun/ballistic/automatic/m90/afterattack(atom/target, mob/living/user, flag, params)
-	if(select == 2)
+	var/list/modifiers = params2list(params)
+	if(modifiers["right"] && !Adjacent(target))
```
Use the new alt attack procs, and don't bother checking adjacency.

TJatPBnJ replied:
I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

TJatPBnJ replied:
After staring at a screen confused for several hours, I've finally found that the alt attack procs aren't even called when a gun is fired. I feel slightly less sane, thanks.

TJatPBnJ replied:
> I was checking adjacency so you could still hold people up with the gun without blowing yourself up, are you sure?

The staring has also made me realize why I didn't need to check adjacency.

TJatPBnJ committed:
Merge remote-tracking branch 'upstream/master'

TJatPBnJ committed:
Merge branch 'master' into m90-better

TJatPBnJ committed:
use the proc

TJatPBnJ committed:
Revert "use the proc"

This reverts commit a2dcbec457829aed13752f16f3c4948e04217477.

TJatPBnJ committed:
use the proc

it uses the proc

TJatPBnJ committed:
restore sanity

guns can now actually use afterattack_alt, if there's another way i should've done this let me know

Mothblocks reviewed (changes requested):
use _secondary now

TJatPBnJ committed:
seccondary

TJatPBnJ committed:
(SECONDARY)

TJatPBnJ committed:
SECOnnnddary

Jared-Fogle commented on code/modules/projectiles/guns/ballistic/automatic.dm:
```diff
@@ -164,11 +164,8 @@
-	else
-		return ..()
+/obj/item/gun/ballistic/automatic/m90/afterattack_secondary(atom/target, mob/living/user, flag, params)
+	underbarrel.afterattack(target, user, flag, params)
```
This shouldn't return null. You'll find that it'll runtime when you right click someone adjacent to you.

Return `SECONDARY_ATTACK_CONTINUE_CHAIN` instead.

TJatPBnJ committed:
returns

tgstation-server commented:
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.
